docs/designers-developers/developers/backward-compatibility/README.md

```diff
@@ -1 +1,65 @@
+* Changes to arguments (new arguments, modification of semantics) is possible if we guarantee that all previous calls are still possible.
+
+### How to preserve backward compatibility for a React Component
+
+* The name of the component should not change.
+* The props of the components should not be removed.
+* Existing prop values should continue to be supported. If a component accepts a function as prop, we can update the component to accept a new type for the same prop, but it shouldn't break existing usage.
+* Adding new props is allowed.
+* React Context dependencies can only be added or removed if we ensure the previous context contract is not breaking.
+
+### How to preserve backward compatibility for a Block
+
+* Existing usage of the block should not break or be marked as invalid when the editor is loaded.
+* The styling of the existing blocks should be guaranteed.
+* Markup changes should be limited to the minimum but If a block need to change its saved markup making previous versions invalid, a [**deprecated version**](/docs/designers-developers/developers/block-api/block-deprecation.md) for the block should be added.
+
+## Class names and DOM updates
+
+Class names and DOM nodes used inside the tree of React components are not considered part of the public API and can be modified. 
+
+Changes to these should be done with caution as it can affect the styling and behavior of third-party code (Even if they should not rely use these in first place). Keep the old ones if possible. If not, document these changes and write a dev note about the changes.
+
+## Deprecations
+
+As the project evolves, flaws to existing APIs are discovered, or updates are required to support new features. When this happens, we try to guarantee that existing APIs don't break and build new alternative APIs.
+
+To encourage third-party developers to adopt the new APIs instead, we can use the [**deprecated**](/packages/deprecated/README.md) helper to show a message explaining the deprecation and proposing the alternative whenever the old API is used.
+
+## Dev Notes
+
+Dev notes are [posts published on the make/core site](https://make.wordpress.org/core/tag/dev-notes/) prior to WordPress releases to inform third-party developers about important changes to the developer APIs, these changes can include:
+* Promote new APIs.
+* Explain how some changes to existing APIs might affect existing plugins and themes (Example: classname changes...)
+* When breaking backward compatibility is unavoidable, explain the reasons and the migration flow.
+* When important deprecations are introduced (even without breakage), explain why and how third-party developers can update their code base in consequence.
+
+### Dev Note Workflow
+
+* When working on a pull request and the need for a dev note is discovered, add the **Needs Dev Note** label to the PR.
+* If possible, add a comment to the PR explaining why the dev note is needed.
```

Comment:
```suggestion
* If possible, add a comment to the PR explaining why the Dev Note is needed.
```

Reply:
I'm going to discard the dev note suggestions per @ellatrix's suggestion it's probably not that important anyway.

Reply:
Yeah, let's keep it lowercase.

https://www.grammarly.com/blog/capitalization-rules/